dapp/test/crowdsale.js

```diff
@@ -56,6 +57,44 @@ class TestBosonRunner {
     }
 }
 
+// Get Ether balance of a given account. Why does this not exist in
+// truffle-contract?
+async function getBalance(account) {
+    return new Promise((resolve, reject) => {
+        // Ugh.
+        const web3 = Crowdsale.web3;
+        web3.eth.getBalance(account, (error, balance) => {
+            if (error) {
+                reject(error);
+            } else {
+                resolve(balance);
+            }
+        });
+    });
+}
+
+// Calculate the price we paid in ethers for a given transaction receipt.
+async function gasUsedPrice(receipt) {
+    const gasUsed = receipt.gasUsed;
+    const transaction = await new Promise((resolve, reject) => {
```

Comment:
Might want to use/write some kind of `promisify` function to avoid boilercode errors.
Eg:

```typescript
const getTransaction = promisify(Crowdsale.web3.eth.getTransaction);
const gasPrice = (await getTransaction(receipt.transactionHash)).gasPrice;
```

Reply:
It was a conscious decision from my side to not do something like that - tests should be easy to read and edit. From my experience, introducing abstraction in tests sometimes leads to them being treated like sacred cows and thus stagnating or causing hairiness of further code added. Sometimes it's better to Write Everything Twice then Don't Repeat Yourself...